### PR TITLE
Core Library with Audit Entity Listener, Soft Delete and Created/Modified by and on  

### LINKED ISSUE #4: Core package setup

### LINKED ISSUE #5: Base framework setup

### PR DESCRIPTION
**Pull request by harshadk-sourcefuse**
Core Library with Audit Entity Listener, Soft Delete and Created/Modified by and on

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,5 @@ build/
 .vscode/
 
 .DS_Store
+
+/node_modules/
```
Comment (samarpan-b) [inline .gitignore]: not needed

File: libs/core/pom.xml [hunk 2]
```diff
@@ -60,6 +60,19 @@
 			<artifactId>spring-security-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+		    <groupId>com.h2database</groupId>
+		    <artifactId>h2</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.postgresql</groupId>
```
Comment (samarpan-b) [inline libs/core/pom.xml]: why do we need postgresql in core ? it should not depend on any DB. Service should depend on DB.

File: libs/core/src/main/java/com/sourcefuse/jarc/core/constants/Constants.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.sourcefuse.jarc.core.constants;
+
+public class Constants {
```
Comment (samarpan-b) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/constants/Constants.java]: Just keep enum in this file. No need to have constants class.

File: libs/core/src/main/java/com/sourcefuse/jarc/core/entitylisteners/AuditLogEntityListener.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        .getPrincipal();
+      AuditLog auditLog = new AuditLog();
+      auditLog.setAction(action);
+      auditLog.setActedOn(entity.getTableName());
+      auditLog.setActionKey(entity.getClass().getSimpleName() + "_Logs");
+      auditLog.setBefore(before);
+      auditLog.setAfter(after);
+      auditLog.setEntityId(entity.getId());
+      auditLog.setActor(currentUser.getUser().getId());
+      log.info("::: audit Log {}", auditLog.toString());
```
Comment (samarpan-b) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/entitylisteners/AuditLogEntityListener.java]: which logger is this ? Can we standardize a log format for all logs ?

File: libs/core/src/main/java/com/sourcefuse/jarc/core/models/base/SoftDeleteEntity.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@MappedSuperclass
+@EntityListeners(SoftDeleteEntityListner.class)
+public abstract class SoftDeleteEntity extends BaseEntity {
+
+  @Column(columnDefinition = "boolean default false")
+  boolean deleted;
+
+  Date deletedOn;
```
Comment (samarpan-b) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/models/base/SoftDeleteEntity.java]: dont we need @column annotation for these ?

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.repository.NoRepositoryBean;
+
+@NoRepositoryBean
+public interface SoftDeletesRepository<
```
Comment (samarpan-b) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: Shouldn't we create a repositories folder and put this into that ?

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  T extends SoftDeleteEntity, ID extends Serializable
+>
+  extends JpaRepository<T, ID> {
+  List<T> findAllActive();
```
Comment (samarpan-b) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: We should override existing findAll functions. Not create new ones.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "j-arc",
```
Comment (samarpan-b) [inline package.json]: why this ?

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: libs/core/src/main/java/com/sourcefuse/jarc/core/entitylisteners/AuditLogEntityListener.java (hunk 4 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/entitylisteners/AuditLogEntityListener.java]: logger is slf4j, can you provide the log format will change the format for all logs

File: package.json (hunk 8 above)
Comment (harshadk-sourcefuse) [reply package.json]: added prettier , should i remove the name key from package json?

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java (hunk 6 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: Done

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java (hunk 7 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: Made changes as per discussion

File: package.json (hunk 8 above)
Comment (harshadk-sourcefuse) [reply package.json]: Removed this file, will look into how to do this using mvn commands

File: libs/core/pom.xml (hunk 2 above)
Comment (harshadk-sourcefuse) [reply libs/core/pom.xml]: removed this dependency

File: libs/core/src/main/java/com/sourcefuse/jarc/core/constants/Constants.java (hunk 3 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/constants/Constants.java]: Done

File: .gitignore (hunk 1 above)
Comment (harshadk-sourcefuse) [reply .gitignore]: removed this

File: libs/core/src/main/java/com/sourcefuse/jarc/core/models/base/SoftDeleteEntity.java (hunk 5 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/models/base/SoftDeleteEntity.java]: You don't need to specify @Column annotation to persist a bean property. we need `@Column` field to specify only columnDefination, or if the column is not nullable, or to change the name that should reflect on db ..etc, @Entity Handles the rest

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**